### PR TITLE
[3006.x] Prevent MasterMinon from loading config from /etc/salt/minion.d/*conf

### LINKED ISSUE #64219: [BUG] MasterMinion loads config from master config file, but includes from minion.d/
**Description**
The MasterMinion class (in `salt/minion.py`) loads its configuration from 

* /etc/salt/master
* /etc/salt/master.d/*.conf
* /etc/salt/minion.d/*.conf

Note that it does **not** load config from /etc/salt/minion

**Setup**

Setup to demonstrate the issue. 

* Uses `auth_tries` as an arbitrary config item that is minion only and won't affect the tests

* Test orchestration:
```yaml
# /srv/salt/test_mminion.sls
test_mminion:
  test.show_notification:
    - text: "Value set in {{ salt['pillar.get']('set_in_file') }} to {{ salt['pillar.get']('value') }}, value from config: {{ salt['config.get']('auth_tries') }}"

```

* Script to run through setting value in the four possible places (/etc/salt/master, /etc/salt/master.d, /etc/salt/minion, /etc/salt/minion.d) and call the orchestration:

```shell
!/bin/bash

# Set in master config
echo "auth_tries: 11" >> /etc/salt/master
salt-master -d
sleep 5s
salt-run state.orchestrate test_mminion pillar='{"set_in_file": "etc/salt/master", "value": "11"}'
pkill -f salt-master
sed -i  "s/auth_tries:.*//" /etc/salt/master

# Set in minion config
echo "auth_tries: 22" >> /etc/salt/minion
salt-master -d
sleep 5s
salt-run state.orchestrate test_mminion pillar='{"set_in_file": "etc/salt/minion", "value": "22"}'
pkill -f salt-master
sed -i  "s/auth_tries:.*//" /etc/salt/minion

# Set in master.d
echo "auth_tries: 33" >> /etc/salt/master.d/test.conf
salt-master -d
sleep 5s
salt-run state.orchestrate test_mminion pillar='{"set_in_file": "etc/salt/master.d/test.conf", "value": "33"}'
pkill -f salt-master
rm /etc/salt/master.d/test.conf

# Set in minion.d
echo "auth_tries: 44" >> /etc/salt/minion.d/test.conf
salt-master -d
sleep 5s
salt-run state.orchestrate test_mminion pillar='{"set_in_file": "etc/salt/minion.d/test.conf", "value": "44"}'
pkill -f salt-master
rm /etc/salt/minion.d/test.conf

```

**Steps to Reproduce the behavior**

## Method 1

Run script above. Output shows that value for `auth_tries` will be read from `/etc/salt/minion.d/*.conf` as well as `/etc/salt/master` and `/etc/salt/master.d/*.conf`

Example output:

```yaml
$ ./test_mminion.sh
test_master:
----------
          ID: test_mminion
    Function: test.show_notification
      Result: True
     Comment: Value set in etc/salt/master to 11, value from config: 11
     Changes:

Summary for test_master
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
test_master:
----------
          ID: test_mminion
    Function: test.show_notification
      Result: True
     Comment: Value set in etc/salt/minion to 22, value from config: 7
     Changes:

Summary for test_master
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
test_master:
----------
          ID: test_mminion
    Function: test.show_notification
      Result: True
     Comment: Value set in etc/salt/master.d/test.conf to 33, value from config: 33
     Changes:

Summary for test_master
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
test_master:
----------
          ID: test_mminion
    Function: test.show_notification
      Result: True
     Comment: Value set in etc/salt/minion.d/test.conf to 44, value from config: 44
     Changes:

Summary for test_master
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
```

## Method 2

* Use a debugger
* Set a break point at line 118 in salt/runners/state.py
* Set a value (eg `auth_tries`) in `/etc/salt/minion.d/test.conf`
* Run an orchestration
* Inspect `minion.opts['auth_tries']` when you hit the breakpoint

**Expected behavior**

Do not expect the MasterMinion class to read config from `/etc/salt/minion.d/*.conf`, only expect it to read from `/etc/salt/master` and `/etc/salt/master.d/*.conf`

**Versions Report**
```yaml
Salt Version:
          Salt: 3005.1+3093.g4fc4450bf0

Python Version:
        Python: 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0]

Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: 2.8.2
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.1.2
       libgit2: Not Installed
  looseversion: 1.1.2
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.5
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     packaging: 23.0
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.17
        pygit2: Not Installed
  python-gnupg: Not Installed
        PyYAML: 6.0
         PyZMQ: 25.0.2
        relenv: 0.11.2
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: ubuntu 22.04.2 jammy
        locale: utf-8
       machine: x86_64
       release: 5.15.0-69-generic
        system: Linux
       version: Ubuntu 22.04.2 jammy
```
</details>

**Additional context**

* Issue was originally reported in https://github.com/saltstack/salt/issues/64158 which surfaced the problem due to the changes in https://github.com/saltstack/salt/pull/64037 changing the `salt-master` process to run as a non-root user

* Fix for that is in https://github.com/saltstack/salt/pull/64174 - that fixes the immediate issue by making sure `/etc/salt/minion.d` is readable by the `salt` user. That's a sensible approach to do the minimum bugfix, but we should resolve the issue of the master reading config from `/etc/salt/minion.d`

* I'm pretty sure this is unintended behaviour - if the intent was to merge in the normal minion config to the config for the MasterMinion, it would read from `/etc/salt/minion` as well as `/etc/salt/minion.d/*.conf`

*  This has probably been around for a long time - looks like it was spotted but not fixed in https://github.com/saltstack/salt/pull/30068 back in 2015

* I'm of the opinion that the MasterMinion should **only** read config from `/etc/salt/master` and `/etc/salt/master.d/*.conf`. Sharing just include config with a normal minion on the same box does not make sense. This behaviour is also undocumented and likely to cause unexpected outcomes.

* As this behaviour has been in place for a long time, it's going to need clear communication about it being fixed as some people may rely on it.

### PR DESCRIPTION
**Pull request by barneysowood**
### What does this PR do?
Stops the MasterMinion loading config from  `/etc/salt/minion.d/*.conf` by ensuring `default_include` is set to `DEFAULT_MASTER_OPTS["default_include"] ` when `minion_config()` is called from `mminion_config()`

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/64219

### Previous Behavior
MasterMinion class would load config from `/etc/salt/minion.d/*.conf`

### New Behavior
MasterMinion class only loads config from `/etc/salt/master` and `/etc/salt/master.d/*.conf`

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes/No

### CODE REVIEW

**Comment by barneysowood** [comment]
Todo:

* Add tests
* Work out if we should also fix code in `minion_config()` that handles env vars and assumes minion config paths.

**Review by dwoz — DISMISSED** [review]

**Comment by Ch3LL** [comment]
@barneysowood this is still in draft mode. We hope to release the next bug fix really soon, mid August. Is this close to being ready?

**Comment by barneysowood** [comment]
@Ch3LL - let me know if tests for this look reasonable.

I'm also not sure if we want some more documentation of the change beyond the changelog entry - let me know what you think.

**Comment by barneysowood** [comment]
Note I'll need to add some additional changes to this to remove the various bits that change ownership of `/etc/salt/minion.d` in #64194 before this is merged - that'll require that PR having be merged to 3006.x so I can rebase on to that first though.

**Comment by Ch3LL** [comment]
K, our code freeze is tonight, so if we can get it in we will. For now I'll atleast give it a review as the code is so we can be proactive.

**Review by Ch3LL — APPROVED** [review]

**Comment by barneysowood** [comment]
Created #65001 - which can be merged once this and #64194 have been merged to 3006.x

**Review by garethgreenaway — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**
